src/pages/ReimbursementAccount/ACHContractStep.js

```diff
@@ -36,74 +38,106 @@ const propTypes = {
+                    errors[`beneficialOwner.${ownerID}.city`] = this.props.translate('bankAccount.error.addressCity');
+                }
+
+                if (!ValidationUtils.isRequiredFulfilled(values[`beneficialOwner.${ownerID}.state`])) {
+                    errors[`beneficialOwner.${ownerID}.state`] = this.props.translate('bankAccount.error.addressState');
+                }
+
+                if (!ValidationUtils.isRequiredFulfilled(values[`beneficialOwner.${ownerID}.zipCode`])
+                    || !ValidationUtils.isValidZipCode(values[`beneficialOwner.${ownerID}.zipCode`])) {
+                    errors[`beneficialOwner.${ownerID}.zipCode`] = this.props.translate('bankAccount.error.zipCode');
+                }
+            });
         }
 
-        const errors = {};
-        _.each(this.requiredFields, (inputKey) => {
-            if (ValidationUtils.isRequiredFulfilled(this.state[inputKey])) {
-                return;
-            }
+        if (!ValidationUtils.isRequiredFulfilled(values.acceptTermsAndConditions)) {
+            errors.acceptTermsAndConditions = this.props.translate('common.error.acceptedTerms');
+        }
 
-            errors[inputKey] = true;
-        });
-        BankAccounts.setBankAccountFormValidationErrors({...errors, beneficialOwnersErrors});
-        return _.every(beneficialOwnersErrors, _.isEmpty) && _.isEmpty(errors);
+        if (!ValidationUtils.isRequiredFulfilled(values.certifyTrueInformation)) {
+            errors.certifyTrueInformation = this.props.translate('beneficialOwnersStep.error.certify');
+        }
+
+        return errors;
     }
 
-    removeBeneficialOwner(beneficialOwner) {
+    removeBeneficialOwner(ownerID) {
         this.setState((prevState) => {
-            const beneficialOwners = _.without(prevState.beneficialOwners, beneficialOwner);
+            const beneficialOwners = _.without(prevState.beneficialOwners, ownerID);
 
             // We set 'beneficialOwners' to null first because we don't have a way yet to replace a specific property without merging it.
             // We don't use the debounced function because we want to make both function calls.
-            BankAccounts.updateReimbursementAccountDraft({beneficialOwners: null});
-            BankAccounts.updateReimbursementAccountDraft({beneficialOwners});
+            FormActions.setDraftValues(ONYXKEYS.REIMBURSEMENT_ACCOUNT, {beneficialOwners: null});
+            FormActions.setDraftValues(ONYXKEYS.REIMBURSEMENT_ACCOUNT, {beneficialOwners});
 
-            // Clear errors
-            BankAccounts.setBankAccountFormValidationErrors({});
             return {beneficialOwners};
         });
     }
 
     addBeneficialOwner() {
-        this.setState(prevState => ({beneficialOwners: [...prevState.beneficialOwners, {}]}));
+        this.setState((prevState) => {
+            const beneficialOwners = [...prevState.beneficialOwners, NumberUtils.rand64()];
```

Comment:
What's the `rand64()` doing here?

Reply:
Context: https://github.com/Expensify/App/pull/13501#discussion_r1046457172

Reply:
Can we summarize it and if so, is it something we could leave a comment in the code about? 

I am certain I would not find that conversation if I came across this and wondered what it was for.